lib/ace/ext/papeeria/katex-previewer.coffee

```diff
@@ -1,65 +1,10 @@
 define((require, exports, module) ->
   LatexParsingContext = require("ace/ext/papeeria/latex_parsing_context")
-  exports.setupPreviewer = (editor, popoverHandler) ->
-    katex = null
-    popoverHandler ?= {
-      options: {
-        html: true
-        placement: "bottom"
-        trigger: "manual"
-        title: "Formula"
-        container: editor.container
-      }
+  TokenIterator = require("ace/token_iterator").TokenIterator
+  Range = require("ace/range").Range
 
-      show: (jqPopoverContainer, content, position) ->
-        jqPopoverContainer.css(position)
-        popoverHandler.options.content = content
-        jqPopoverContainer.popover(popoverHandler.options)
-        jqPopoverContainer.popover("show")
-        return
-
-      destroy: (jqPopoverContainer) ->
-        jqPopoverContainer.popover("destroy")
-
-      popoverExists: (jqPopoverContainer) ->
-        jqPopoverContainer.data()?.popover?
-
-      setContent: (jqPopoverContainer, content) ->
-        jqPopoverElement = jqPopoverContainer.data().popover.tip().children(".popover-content").html(content)
-
-      setPosition: (jqPopoverContainer, position) ->
-        jqPopoverElement = jqPopoverContainer.data().popover.tip().css(position)
-    }
-
-    initKaTeX = (onLoaded) ->
-      # Adding CSS for demo formula
-      cssDemoPath = require.toUrl("./katex-demo.css")
-      linkDemo = $("<link>").attr(
-        rel: "stylesheet"
-        href: cssDemoPath
-      )
-      $("head").append(linkDemo)
-
-      # Adding DOM element to place formula into
-      span = $("<span>").attr(
-        id: "formula"
-      )
-      $("body").append(span)
-
-      require(["ace/ext/katex"], (katexInner) ->
-        katex = katexInner
-        onLoaded()
-        return
-      )
-      return
-
-    jqEditorContainer = $(editor.container)
-    getFormulaElement = -> $("#formula")
-    KATEX_OPTIONS = {displayMode: true, throwOnError: false}
-    TokenIterator = require("ace/token_iterator").TokenIterator
-    Range = require("ace/range").Range
-
-    erh = EquationRangeHandler = {
+  getEquationRangeHandler = (editor) ->
```

Comment:
@dbarashev To test `EquationRangeHandler`, I had to carry it out this way. Initially, I didn't want to make it a class, because classes are, AFAIK, heavyweight and have a lot of overhead. If I wrap it all in a function like that, is it still better, than just making it a class?